app/models/language-settings-store/language-settings-store.ts

```diff
@@ -0,0 +1,46 @@
+import { Instance, SnapshotOut, types } from "mobx-state-tree"
+import i18n from "i18n-js"
+
+import { LANGUAGES, SYSTEM_LANGUAGE } from "../../i18n"
+
+/**
+ * Language settings.
+ */
+export const LanguageSettingsStoreModel = types
+  .model("LanguageSettingsStore")
+  .props({
+    activeLanguageKey: types.optional(types.string, SYSTEM_LANGUAGE),
+  })
+  .views(self => ({
+    get items() {
+      return LANGUAGES.filter(language => language.canSet).map(language => ({
+        ...language,
+        isActive: self.activeLanguageKey === language.key,
+      }))
+    },
+  }))
+  .actions(self => {
+    let onChangeCallback: (languageKey?: string) => void
+
+    return {
+      afterCreate() {
+        i18n.locale = self.activeLanguageKey
+      },
+      listenChange(callback: (languageKey?: string) => void) {
+        onChangeCallback = callback
+      },
+      setLanguage(languageKey: string) {
+        self.activeLanguageKey = languageKey
+        i18n.locale = languageKey
+        if (onChangeCallback) onChangeCallback(languageKey)
```

Comment:
Should we add try catch here?

Reply:
you mean try catch the `onChangeCallback`?

Reply:
yes, very minor though